apps/ledger-live-mobile/src/components/RequiresLocation/hooks/useAndroidEnableLocation.ts

```diff
@@ -0,0 +1,98 @@
+import { useCallback, useEffect, useState } from "react";
+import { NativeEventEmitter, NativeModules, Platform } from "react-native";
+import LocationHelperModule, {
+  CheckAndRequestResponse,
+} from "../../../native-modules/LocationHelperModule";
+
+/**
+ * Hook returning a callback that checks the location services state and requests (if necessary) the user to enable them
+ *
+ * If you want to enable the location services directly and/or keep track of the location services state,
+ * use useAndroidEnableLocation defined below.
+ *
+ * Only available on Android.
+ *
+ * @returns the callback
+ */
+export function useAndroidPromptEnableLocationCallback() {
+  return useCallback(async (): Promise<CheckAndRequestResponse> => {
+    try {
+      if (Platform.OS === "android") {
+        const response =
+          await LocationHelperModule.checkAndRequestEnablingLocationServices();
+        return response;
+      }
+
+      return "ERROR_UNKNOWN";
+    } catch (e) {
+      // Should never be reached, a checkAndRequestEnablingLocationServices only resolves
```

Comment:
never reached ... unless the native `checkAndRequestEnablingLocationServices` gets removed/renamed in which case you'll get a proper error.
Since there is no straightforward way today to enforce that the contract is respected (that the method is implemented on the native side), I think we should at least log the error (probably even throw it)

Reply:
Yes you're right - i'm not going to throw here, because then i would try/catch where it is used below, and do basically the same (`setLocationServicesState("disabled")` + log)

So i'm going to log the error, add a new possible returned value `"LOCATION_MODULE_THREW_ERROR"` and return this